### PR TITLE
Adds a new evolved child zombie

### PR DESCRIPTION
**Pull request by DoctorBoomstick**
#### Summary
Content "Adds the swollen wretch as a new possible child zombie upgrade, plus fungus variant."

#### Purpose of change
This started life as a short joke concept on the discord chat, before I thought it’d be a genuinely possibly cool idea to add a new flying boomer-like zombie to the game, and subsequently took it upon myself to add this monster as an upgrade.

#### Describe the solution
I’ve added a zombie child upgrade called the swollen wretch, which is something of an amalgamation between a boomer zombie, a pupating zombie, and a regular child zombie. More than anything it’s a floating flesh raptor incubator, which traps decomposition caused gasses within itself, causes a chemical chain reaction where in the gas is burned within it and allows it to obtain a measure.of buoyancy, thus giving it a reason for why it can fly. It’s speed and health are very low, with the main threats being the large amount of flesh raptors it harbours within itself. In addition, I’ve also added it’s fungal form and the fungal flesh raptors that it spawns. I’ve included the regular zombie form of the wretch into the zombie upgrades file, under the child zombie grouping.

#### Describe alternatives you've considered
Letting a joke remain as a joke and not deciding to Frankenstein up this creature.

#### Testing
Spawned in a bunch of iterations of the creature, checked to see that all their properties function correctly, spawned in a few zombie children and a zombie master to observe if they would mutate into the wretch, and ran the same style of tests on the fungal form.

#### Additional context
No response

### CODE REVIEW

File: data/json/monsters/fungus_zombie.json [hunk 1]
```diff
@@ -261,5 +261,99 @@
     "death_function": { "effect": { "id": "death_fungus", "hit_self": true } },
     "burn_into": "mon_zombie_scorched",
     "flags": [ "SEES", "HEARS", "SMELLS", "STUMBLES", "WARM", "BASHES", "GROUP_BASH", "POISON", "NO_BREATHE", "PUSH_MON", "FILTHY" ]
+  },
+  {
+    "id": "mon_fungal_wretch",
+    "type": "MONSTER",
+    "name": { "str": "fungal wretch" },
```
Comment (BrettDong) [inline data/json/monsters/fungus_zombie.json:268]:
```suggestion
    "name": { "str": "fungal wretch", "str_pl": "fungal wretches" },
```

File: data/json/monsters/fungus_zombie.json [hunk 2]
```diff
@@ -261,5 +261,99 @@
...
+      "GROUP_BASH",
+      "NO_BREATHE",
+      "FILTHY",
+      "HIT_AND_RUN",
+      "SWARMS",
+      "FLIES"
+    ]
   }
-]
+]
```
Comment (BrettDong) [inline data/json/monsters/fungus_zombie.json:359]: Please restore the newline at the end of the file.

File: data/json/monsters/zed_children.json [hunk 3]
```diff
@@ -261,5 +261,59 @@
     "fungalize_into": "mon_zombie_child_fungus",
     "flags": [ "SEES", "HEARS", "SMELLS", "STUMBLES", "WARM", "GUILT", "POISON", "NO_BREATHE", "REVIVES", "HARDTOSHOOT", "FILTHY" ],
     "//": "GUILT because it still looks enough like a child to evoke pity"
+  },
+  {
+    "id": "mon_zombie_wretch",
+    "type": "MONSTER",
+    "name": { "str": "swollen wretch" },
```
Comment (BrettDong) [inline data/json/monsters/zed_children.json:268]:
```suggestion
    "name": { "str": "swollen wretch", "str_pl": "swollen wretches" },
```

File: data/json/monsters/fungus_zombie.json [hunk 4]
```diff
@@ -261,5 +261,106 @@
     "death_function": { "effect": { "id": "death_fungus", "hit_self": true } },
     "burn_into": "mon_zombie_scorched",
     "flags": [ "SEES", "HEARS", "SMELLS", "STUMBLES", "WARM", "BASHES", "GROUP_BASH", "POISON", "NO_BREATHE", "PUSH_MON", "FILTHY" ]
+  },
+  {
+    "id": "mon_fungal_wretch",
+    "type": "MONSTER",
+    "name": { "str": "fungal wretch", "str_pl": "fungal wretches" },
+    "description": "Were it not for the head, its orifices sealed by clumps of fungal growth, which limply crowns this massively swollen corpse, it’d be impossible to tell that this was once a human, let alone a child or adolescent.  The beast’s skin, stretched to paper-thin levels by what you can only assume to be lighter than air decomposition caused gasses built up within it, has turned into a spongy fungal covering with intermittent hair-line cracks which let out spores.  Like a nightmarish balloon the abomination is suspended off the floor with a faint glow emanating from within its guts, more than likely a result of the burning putrid gasses which keep it afloat.  At irregular intervals, you can see it’s form bulge and shift, a hint of something moving within it’s mass.",
```
Comment (TheShadowFerret) [inline data/json/monsters/fungus_zombie.json:269]:
```suggestion
    "description": "Were it not for the head, its orifices sealed by clumps of fungal growth, which limply crowns this massively swollen corpse, it’d be impossible to tell that this was once a human, let alone a child or adolescent.  The beast’s skin, stretched to paper-thin levels by what you can only assume to be lighter than air decomposition caused gasses built up within it, has turned into a spongy fungal covering with intermittent hair-line cracks which let out spores.  Like a nightmarish balloon the abomination is suspended off the floor with a faint glow emanating from within its guts, more than likely a result of the burning putrid gasses which keep it afloat.  At irregular intervals, you can see its form bulge and shift, a hint of something moving within its mass.",
```

File: data/json/monsters/fungus_zombie.json [hunk 5]
```diff
@@ -261,5 +261,106 @@
...
+    "default_faction": "fungus",
+    "bodytype": "blob",
+    "species": [ "FUNGUS" ],
+    "volume": "120000 ml",
+    "weight": "45000 g",
+    "//": "Weight is more than it’s zombie counterpart cause of the fungal elements. It’s still rather slow, but it also has more health cause the fungus acts as a sort of makeshift armour.",
```
Comment (TheShadowFerret) [inline data/json/monsters/fungus_zombie.json:275]:
```suggestion
    "//": "Weight is more than its zombie counterpart cause of the fungal elements. It’s still rather slow, but it also has more health cause the fungus acts as a sort of makeshift armour.",
```

**Comment by DoctorBoomstick** [comment]
I mean, my good sir. There isn’t anything which says that we can’t slowly build up a well-stocked kindergarten’s assembly of mutated kids. I’m just going to let some time pass for this PR and my other ones to hopefully get merged, but I’ll keep that idea in the back pocket.

**Comment by I-am-Erk** [comment]
Seems good but I have no idea why this would spawn *more* raptors than a pupating zombie, it's a smaller source and apparently also able to float.

**Comment by DoctorBoomstick** [comment]
That’s fair. The original thought process was to make them spawn more raptors cause, number one, they’re around 120000 ml in volume, which is quite a chunky amount, and number two, to make them a bit more threatening cause as it stands, their only real source of danger stems from the raptors within them. Though on that note, I suppose it might be a bit too over the top, nothing which can’t be fixed by a quick commit.

**Pull request merged**